Gemfile

```
@@ -13,7 +13,34 @@ gem 'bundler'
 gem 'rake'
 
 group :system_tests do
-  gem 'beaker'
+  beaker_gem_options = ENV.fetch('BEAKER_GEM_OPTIONS', ['>= 4.17.0', '< 5.0.0'])
+
+  if "#{beaker_gem_options}".include?(':')
+    # Just pass in BEAKER_GEM_OPTIONS as a string that would represent the usual
+    # hash of options.
+    #
+    # Something like: BEAKER_GEM_OPTIONS=':git => "https://my.repo/beaker.git", :tag => "1.2.3"'
+    #
+    # No, this isn't robust, but it's not really an 'every day' sort of thing
+    # and safer than an `eval`
+    begin
+      gem 'beaker', Hash[
+        beaker_gem_options.split(',').map do |x| # Split passed options on k/v pairs
+          x.strip.split(/:\s|\s+=>\s+/) # Allow for either format hash keys
```

```suggestion
          x.gsub('"', '').strip.split(/:\s|\s+=>\s+/) # Allow for either format hash keys
```

When double quotes like in the example comments, this fails with an `URI::InvalidURIError`.  

```sh
BEAKER_GEM_OPTIONS=':git => "https://github.com/puppetlabs/beaker", :tag => "4.23.1"' bundle
```

However, it works without quotes:
```sh
BEAKER_GEM_OPTIONS=':git => https://github.com/puppetlabs/beaker, :tag => 4.23.1' bundle
```

The suggested code is a brute-force fix for that problem.